git/libgit2/clone.go

```diff
@@ -79,6 +79,14 @@ func (l *Client) cloneBranch(ctx context.Context, url, branch string, opts git.C
 		return nil, fmt.Errorf("unable to fetch remote '%s': %w", url, gitutil.LibGit2Error(err))
 	}
 
+	isEmpty, err := l.repository.IsEmpty()
```

Comment:
Can we add a test case for this as well?

Reply:
kindly see the PR description regarding this

Reply:
We could still have a test case for it, which has a different expectation from GoGit. The key point here was to have something to identify regression in case changes are made here and empty clones would not break again.

As we may decomission the entire libgit2 implementation by Bootstrap GA, we can park this.